src/Bicep.LangServer/Handlers/BicepDefinitionHandler.cs

```diff
@@ -2,58 +2,244 @@
+            {
+                var semanticModel = context.Compilation.GetEntrypointSemanticModel();
+
+                // Find the underlying VariableSyntax being accessed
+                var syntax = result.Origin;
+                var propertyAccesses = new List<string>();
+                while (syntax is PropertyAccessSyntax propertyAccessSyntax)
+                {
+                    propertyAccesses.Insert(0, propertyAccessSyntax.PropertyName.IdentifierName);
+                    syntax = propertyAccessSyntax.BaseExpression;
+                }
+
+                if (syntax is VariableAccessSyntax ancestor
+                    && semanticModel.GetSymbolInfo(ancestor) is DeclaredSymbol ancestorSymbol)
+                {
+                    // If the symbol is a module, we need to redirect the user to the module file
+                    // note: module.name doesn't follow this: it should refer to the declaration of the module in the current file
+                    if (ancestorSymbol.DeclaringSyntax is ModuleDeclarationSyntax moduleDeclarationSyntax
+                    && propertyAccesses.Count == 2)
+                    {
+                        return GetModuleSymbolLocationAsync(request, result, context, moduleDeclarationSyntax, propertyAccesses[0], propertyAccesses[1]);
+                    }
+
+                    // Otherwise, we redirect user to the specified module, variable, or resource declaration
+                    if (GetObjectSyntaxFromDeclaration(ancestorSymbol.DeclaringSyntax) is ObjectSyntax objectSyntax
+                        && ObjectSyntaxExtensions.SafeGetPropertyByNameRecursive(objectSyntax, propertyAccesses.ToArray()) is ObjectPropertySyntax resultingSyntax)
+                    {
+                        return Task.FromResult(new LocationOrLocationLinks(new LocationOrLocationLink(new LocationLink
+                        {
+                            OriginSelectionRange = result.Origin.ToRange(result.Context.LineStarts),
+                            TargetUri = request.TextDocument.Uri,
+                            TargetRange = resultingSyntax.ToRange(result.Context.LineStarts),
+                            TargetSelectionRange = resultingSyntax.ToRange(result.Context.LineStarts)
+                        })));
+                    }
+                }
                 return Task.FromResult(new LocationOrLocationLinks());
             }
 
-            if (result.Symbol is DeclaredSymbol declaration)
+            return Task.FromResult(new LocationOrLocationLinks());
+        }
+
+        private Task<LocationOrLocationLinks> GetModuleSymbolLocationAsync(
+            DefinitionParams request,
+            SymbolResolutionResult result,
+            CompilationContext context,
+            ModuleDeclarationSyntax moduleDeclarationSyntax,
+            string propertyType,
+            string propertyName)
+        {
+            if (moduleDeclarationSyntax.TryGetPath() is StringSyntax pathSyntax
```

Comment:
Any way to simplify this?

Reply:
`pathSyntax.TryGetLiteralValue()` should do most of what you need. 

Btw, getting the literal value involves more than just dropping the quotes from the string token. Bicep strings support various escape sequences that need to be unescaped. The function should handle it already.